File: C++/WEEK 1/SECTION 2/BOY/cropped_boy.cpp

#include <opencv2/opencv.hpp>
#include <opencv2/core.hpp>
#include <opencv2/highgui.hpp>

using namespace std;
using namespace cv;

int main (){


Mat image = imread("/home/paulstein/opencv/Example_course/C++/WEEK 1/SECTION 2/BOY/boy.jpg",1);
Mat copiedImage = image.clone();
//PART1
// Crop out a rectangle
// x coordinates = 170 to 320
// y coordinates = 40 to 200
//Mat crop = image(Range(40,200),Range(170,320));
//imwrite("/home/paulstein/opencv/Example_course/C++/WEEK 1/SECTION 2/BOY/crop.png",crop);


//PART2
//copy the face to the left and to the right
//Mat copyRoi = image(Range(40,200),Range(180,320));
// Find height and width of the ROI
//int roiHeight = copyRoi.size().height;
//int roiWidth = copyRoi.size().width;
// Copy to left of Face
//copyRoi.copyTo(copiedImage(Range(40,40+roiHeight),Range(10,10+roiWidth)));
// Copy to right of Face
//copyRoi.copyTo(copiedImage(Range(40,40+roiHeight),Range(330,330+roiWidth)));
//imwrite("/home/paulstein/opencv/Example_course/C++/WEEK 1/SECTION 2/BOY/copiedRegions.png",copiedImage);


//PART3
//resize image
//int resizeDownWidth = 300;
//int resizeDownHeight = 200;
//Mat resizedDown;
//resize(image, resizedDown, Size(resizeDownWidth, resizeDownHeight), INTER_LINEAR);
// Mess up with the aspect ratio
//int resizeUpWidth = 600;
//int resizeUpHeight = 900;
//Mat resizedUp;
//resize(image, resizedUp, Size(resizeUpWidth, resizeUpHeight), INTER_LINEAR);
//imwrite("/home/paulstein/opencv/Example_course/C++/WEEK 1/SECTION 2/BOY/resizedUp.png",resizedUp);
//imwrite("/home/paulstein/opencv/Example_course/C++/WEEK 1/SECTION 2/BOY/resizedDown.png",resizedDown);


//PART4
// Create an empty image of same size as the original
//Mat mask1 = Mat::zeros(image.size(), image.type());
//imwrite("/home/paulstein/opencv/Example_course/C++/WEEK 1/SECTION 2/BOY/mask1.png",mask1);

//mask1(Range(50,200),Range(170,320)).setTo(255);
//imwrite("/home/paulstein/opencv/Example_course/C++/WEEK 1/SECTION 2/BOY/mask1Revised.png",mask1);

//PART5
//Create a mask based on the colors in the image
Mat mask2;
inRange(image, Scalar(0,0,150), Scalar(100,100,255), mask2);
imwrite("/home/paulstein/opencv/Example_course/C++/WEEK 1/SECTION 2/BOY/mask2.png",mask2);
}

File: PYHTON/WEEK 1/SECTION 2/Boy_image_modify/cropped_boy.py
# Import libraries
import cv2
import numpy as np
#from dataPath import DATA_PATH
import matplotlib.pyplot as plt 	 	

import matplotlib

#Read image
image = cv2.imread("/home/paulstein/opencv/Example_course/PYHTON/SECTION 2/Boy_image_modify/boy.jpg")

# Let's see what image we are dealing with
plt.imshow(image[:,:,::-1])

# Crop out a rectangle
# x coordinates = 170 to 320
# y coordinates = 40 to 200
#crop = image[40:200,170:320]
#plt.imshow(crop[:,:,::-1])
# -----------------------------------------------------------

##Create a copy from the original
#copiedImage = image.copy()
#plt.imshow(copiedImage[...,::-1])
# ///// CROPPING AND PASTING
#copyRoi = image[40:200,180:320]
# Find height and width of the ROI
#roiHeight,roiWidth = copyRoi.shape[:2]
# Copy to left of Face
#copiedImage[40:40+roiHeight, 10:10+roiWidth] = copyRoi
# Copy to right of Face
#copiedImage[40:40+roiHeight, 330:330+roiWidth] = copyRoi
# Display the output
#plt.figure(figsize=[15,15])
#plt.subplot(121);plt.imshow(image[...,::-1]);plt.title("Original Image");
#plt.subplot(122);plt.imshow(copiedImage[...,::-1]);plt.title("Output Image");

# --------------------------------------------------------

# Set rows and columns
#resizeDownWidth = 300
#resizeDownHeight = 200
#resizedDown = cv2.resize(image, (resizeDownWidth, resizeDownHeight), interpolation= cv2.INTER_LINEAR)

# Mess up with the aspect ratio
#resizeUpWidth = 600
#resizeUpHeight = 900
#resizedUp = cv2.resize(image, (resizeUpWidth, resizeUpHeight), interpolation= cv2.INTER_LINEAR)

#plt.figure(figsize=[15,15])
#plt.subplot(131);plt.imshow(image[:,:,::-1]);plt.title("Original Image")
#plt.subplot(132);plt.imshow(resizedUp[:,:,::-1]);plt.title("Scaled Up Image")
#plt.subplot(133);plt.imshow(resizedDown[:,:,::-1]);plt.title("Scaled Down Image")
# -----------------------------------------------------------------
# Scaling Down the image 1.5 times by specifying both scaling factors
scaleUpX = 1.5
scaleUpY = 1.5

# Scaling Down the image 0.6 times specifying a single scale factor.
scaleDown = 0.6

scaledDown = cv2.resize(image, None, fx= scaleDown, fy= scaleDown, interpolation= cv2.INTER_LINEAR)

scaledUp = cv2.resize(image, None, fx= scaleUpX, fy= scaleUpY, interpolation= cv2.INTER_LINEAR)

plt.figure(figsize=[15,15])
plt.subplot(121);plt.imshow(scaledDown[...,::-1]);plt.title("Scaled Down Image, size = {}".format(scaledDown.shape[:2]));
plt.subplot(122);plt.imshow(scaledUp[...,::-1]);plt.title("Scaled Up Image, size = {}".format(scaledUp.shape[:2]));
plt.show()


